### PR TITLE
[ENG-2622] max character limit for withdrawal justifications.

### PR DESCRIPTION
**Pull request by adlius**
- Ticket: [ENG-2622]
- Feature flag: n/a

## Purpose

Add maximum character limit for withdrawal justifications.

## Summary of Changes

Changed `registries-states/is-public` component to use changeset and changeset validations.

Screenshot from 2021-02-23 11-59-59

## QA Notes

[ENG-2622]: https://openscience.atlassian.net/browse/ENG-2622

### CODE REVIEW

**Review by brianjgeiger — APPROVED** [review]

File: lib/registries/addon/components/registries-states/is-public/component.ts [hunk 1]
```diff
@@ -20,41 +24,59 @@ export default class RegistrationIsPublic extends Component {
...
+            }
         }
     });
 
     didReceiveAttrs() {
+        this.changeset = new Changeset(
+            this.registration,
+            lookupValidator(this.changesetValidation),
+            this.changesetValidation,
+        ) as ChangesetDef;
```
Comment (fabmiz) [inline lib/registries/addon/components/registries-states/is-public/component.ts:79]: curious to know any reasons you didn't use the `buildChangeset` util?

**Comment by fabmiz** [comment]
@adlius could you please add some component tests for this?

**Comment by adlius** [comment]
> @adlius could you please add some component tests for this?

We don't have any existing tests for this component. It is also tricky to test a component in the `registries` engine. We should defer this till we stop using engines.

**Comment by fabmiz** [comment]
@adlius could you add some screenshots at least showing validation errors? It's good to give the reviewer added confidence that comes from passing tests, screenshots or screen capture.

**Comment by adlius** [comment]
@fabmiz screenshots are added.

**Pull request merged**